src/com/google/javascript/jscomp/RewriteDynamicImports.java

```diff
@@ -357,4 +375,82 @@ private void injectWrappingFunctionExtern() {
     externsRoot.addChildToBack(wrappingFunctionDefinition);
     compiler.reportChangeToEnclosingScope(wrappingFunctionDefinition);
   }
+
+  /**
+   * A shallow traversal class for the externs to inject the alias.
+   *
+   * Later passes require the names to be defined. For simple name aliases first check
+   * for an existing definition. Qualified name aliases check for an existing definition of the
+   * root name only.
+   */
+  private class AliasInjectingTraversal extends NodeTraversal.AbstractPreOrderCallback {
+    @Override
+    public boolean shouldTraverse(NodeTraversal t, Node n, Node parent) {
+      checkState(aliasIsValid());
+
+      // If the extern is a simple name and already declared, there is nothing to do
+      if (NodeUtil.isValidSimpleName(alias) && t.getScope().getVar(alias) != null) {
+        return false;
+      }
+
+      // Create the type of the import function
+      JSTypeRegistry registry = compiler.getTypeRegistry();
+      JSType unknownType = registry.getNativeType(JSTypeNative.UNKNOWN_TYPE);
+      JSType promiseType = registry.getNativeType(JSTypeNative.PROMISE_TYPE);
+      TemplatizedType promiseTemplatizedType =
+          registry.createTemplatizedType(promiseType.toObjectType(), ImmutableList.of(unknownType));
+      JSType stringType = registry.getNativeType(JSTypeNative.STRING_TYPE);
+
+      Node externsRoot = compiler.getSynthesizedExternsInputAtEnd().getAstRoot(compiler);
+      // "import" is not a valid JS name and will not parse. However we can manually create it
+      // and inject it into the externs.
+      if (alias.equals("import") || NodeUtil.isValidSimpleName(alias)) {
+        Node aliasNode =
+            astFactory.createFunction(
+                alias,
+                astFactory.createParamList("specifier"),
+                astFactory.createBlock(),
+                registry.createFunctionType(promiseTemplatizedType, stringType));
+        aliasNode.srcrefTree(externsRoot);
+        externsRoot.addChildToBack(aliasNode);
+      } else {
+        String aliasRootName = NodeUtil.getRootOfQualifiedName(alias);
+        Var aliasVar = t.getScope().getVar(aliasRootName);
+        // If the namespace root is defined, just assume the whole name is properly defined.
+        // This may need revisited in the future, but checking if the full qualified name
+        // is properly defined is difficult here. As long as the root of the qualified name is
+        // defined, other passes seem be ok even with missing properties.
+        if (aliasVar != null) {
+          return false;
+        }
+        Node aliasRootNode =
+            astFactory.createSingleVarNameDeclaration(aliasRootName, astFactory.createObjectLit());
+        aliasRootNode.srcrefTree(externsRoot);
+        externsRoot.addChildToBack(aliasRootNode);
+
+        // Define the rest of the parts of the name
+        String[] aliasNameParts = alias.split("\\.");
```

Comment:
We usually avoid String's `.split` method in favor of Guava's `Splitter.on('.')` instead.

Reply:
Easy fix - done.